sunpy/physics/tests/test_differential_rotation.py

```diff
@@ -391,3 +391,21 @@ def test_differential_rotation(aia171_test_map):
     with pytest.warns(UserWarning, match="Using 'time' assumes an Earth-based observer"):
         rot_map = differential_rotate(aia171_test_map, time=2*u.day)
     return rot_map.data
+
+def test_rsun_ref_fallback(aia171_test_map):
+    from sunpy.util.exceptions import SunpyUserWarning
+    # Ensure rsun_ref is initially in the metadata
+    assert 'rsun_ref' in aia171_test_map.meta
+
+    # Remove rsun_ref from metadata
+    del aia171_test_map.meta['rsun_ref']
+    # Perform differential rotation and check for the correct warning
+    with pytest.warns(SunpyUserWarning) as record:
+        rot_map = differential_rotate(aia171_test_map, time=2 * u.day)
+    
+    for warning in record:
+        print(f"Captured warning: {warning}")
+
+    # Check that rsun_ref in the output map is the default value
+    assert 'rsun_ref' in rot_map.meta
+    print(f"rsun_ref value after rotation: {rot_map.meta['rsun_ref']}")
```

Comment:
We don't print in tests, we check that warnings are raised and we check for values of the properties.